meeting-notes/2019/Q2/2019-05-13--js-core-dev-team-weekly.md

```diff
@@ -0,0 +1,168 @@
+    - [libp2p/interface-connection#29#issuecomment-491262150](https://github.com/libp2p/interface-connection/pull/29#issuecomment-491262150)
+  - Updated hapi in `js-libp2p-webrtc-star`
+    - [libp2p/js-libp2p-webrtc-star#173](https://github.com/libp2p/js-libp2p-webrtc-star/pull/173)
+  - interop tests for IPNS over Pubsub green again
+    - [ipfs/interop#39](https://github.com/ipfs/interop/pull/39)
+- Blocked: N/A
+- Next:
+ - get initial version of `js-gossipsub` PR merged 
+ - update `interface-connection` PR with the new proposal and tests for it
+ - create `interface-stream`
+ - Data Terra Nemo
+
+@jimpick
+- Done:
+   - peer-base-pinner - removed IPNS, use metadata from ipfs-cluster instead
+   - "mini projects" brainstorming with Andrew Nesbitt
+   - Sync up with @alanshaw
+   - IPFS Camp prep - Course "C" - Apps with Hugo / Lidel
+   - Portland trip
+     - csv,conf https://csvconf.com/
+       - Ilya Kreymer
+       - Peter van Hardenberg
+       - Simon Willison
+       - Dat Project
+     - also, Gozala, Michael Burns, Jeffrey Yaskin (Google), Kyle Drake
+- Blocked:
+- Next:
+   - More IPFS Camp prep
+   - More work on PeerPad / peer-base-pinner
+
+@hugomrdias
+- Done:
+  - ipns resolve fqdn PRs
+  - ipns over dns resolver
+  - IPFS Camp Course C 
+- Blocked:
+- Next:
+  - Merge File DOM Api PRs
+  - More IPFS Camp 
+  - Attend DTN Berlin
+
+@lidel
+- Done:
+  - Releases
+    - ipfs-companion [v2.8.2](https://github.com/ipfs-shipyard/ipfs-companion/releases/tag/v2.8.2)  (Stable)
+      - includes Prototype of Embedded HTTP Gateway in Brave
+        - opening `/ipns/tr.wikipedia-on-ipfs.org/wiki/Mars.html` works – details in [#719](https://github.com/ipfs-shipyard/ipfs-companion/pull/719) 
+    - is-ipfs  [v0.6.1](https://github.com/ipfs/is-ipfs/releases/tag/v0.6.1)
+    - chrome-net [v3.2.2](https://github.com/feross/chrome-net/pull/38)
+    - IPFS Camp Course C
+- Blocked:
+  - js-ipfs gateway needs to load files/dirs from hamt shards  [js-ipfs-http-response/pull/19](https://github.com/ipfs/js-ipfs-http-response/pull/19)
+      - depends on [js-ipfs-mfs/pull/48](https://github.com/ipfs/js-ipfs-mfs/pull/48)
+  - Rebased [js-ipfs/pull/2020](https://github.com/ipfs/js-ipfs/pull/2020) parked until remaining  IPNS PRs land
+- Next:
+    - IPFS Camp
+    - Companion/Brave
+
+@achingbrain
+- Done:
+  - async/await for unixfs & mfs
+  - IPFS Camp prep - libp2p planning
+- Blocked:
+  - async/await for unixfs & mfs
+- Next:
+  - async/await for unixfs & mfs
+  - DTN Conf
+
+### Questions
+
+- should we do an iterative non-breaking approach for the async iterators in modules such as the DHT?
+  - See the recording for a longer version of the discussion. The short answer is, we'll leave it up to maintainer discression. Some modules, especially those in heavy active development will be difficult to do full async/await changes. Ideally we want to minimize duplication of work where possible, but in some places this may be unavoidable. 
```

Comment:
In general I think we want these same things:

1. Do as little work as possible
1. Have short lived PRs by attacking refactor in priority order (this is basically enable 1.)
1. Don't put off contributors - all contributions are valuable!
1. Don't confuse _new_ contributors with mixed async await and callbacks approach
1. Be pragmatic, not all of these things can always be achieved at the same time 😜

The DHT is a difficult one because a) it is under heavy development, and b) the existing async/await PR was probably a little premature and has had to sit around for a while...._and_ an alternative PR has been proposed already by a contributor that adds an adapter layer.

Given those circumstances I'd probably now spend the time to review the alternative PR and get it merged ASAP (if it's good!) so that we can continue heavy development and not have to keep porting changes in master to an async/await PR.

Reply:
I agree with all these points 👍

I believe the particular PR in question was made by taking my out-of-date PR and backporting it into master. Whereas a better approach might be to start with master and make incremental PRs

Reply:
Good summary Alan! I agree with that approach 👍